From 9bce2dfd506a1ac405f19a4d35bb6735af6fd605 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:35 -0400
Subject: [kernel] perf: Add new conditional branch filter 'PERF_SAMPLE_BRANCH_COND'

Message-id: <1410181752-25631-31-git-send-email-jolsa@redhat.com>
Patchwork-id: 91708
O-Subject: [PATCH RHEL7.1 BZ1134356 030/307] perf: Add new conditional branch filter 'PERF_SAMPLE_BRANCH_COND'
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit bac52139f0b7ab31330e98fd87fc5a2664951050
Author: Anshuman Khandual <khandual@linux.vnet.ibm.com>
Date: Thu May 22 12:50:07 2014 +0530

description
===========
This patch introduces new branch filter PERF_SAMPLE_BRANCH_COND which
will extend the existing perf ABI. This will filter branches which are
conditional. Various architectures can provide this functionality either
with HW filtering support (if present) or with SW filtering of captured
branch instructions.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/uapi/linux/perf_event.h b/include/uapi/linux/perf_event.h
index 42bb5be..87f6492 100644
--- a/include/uapi/linux/perf_event.h
+++ b/include/uapi/linux/perf_event.h
@@ -163,8 +163,9 @@ enum perf_branch_sample_type {
  PERF_SAMPLE_BRANCH_ABORT_TX = 1U << 7, /* transaction aborts */
  PERF_SAMPLE_BRANCH_IN_TX = 1U << 8, /* in transaction */
  PERF_SAMPLE_BRANCH_NO_TX = 1U << 9, /* not in transaction */
+ PERF_SAMPLE_BRANCH_COND  = 1U << 10, /* conditional branches */
 
- PERF_SAMPLE_BRANCH_MAX  = 1U << 10, /* non-ABI */
+ PERF_SAMPLE_BRANCH_MAX  = 1U << 11, /* non-ABI */
 };
 
 #define PERF_SAMPLE_BRANCH_PLM_ALL \
-- 
1.7.1